src/Xamarin.Android.Build.Tasks/Tasks/ReadAndroidManifest.cs

```diff
@@ -64,11 +64,15 @@ public override bool RunTask ()
 				foreach (var uses_library in app.Elements ("uses-library")) {
 					var attribute = uses_library.Attribute (androidNs + "name");
 					if (attribute != null && !string.IsNullOrEmpty (attribute.Value)) {
+						var required = uses_library.Attribute (androidNs + "required")?.Value;
 						var path = Path.Combine (AndroidSdkDirectory, "platforms", $"android-{AndroidApiLevel}", "optional", $"{attribute.Value}.jar");
 						if (File.Exists (path)) {
 							libraries.Add (new TaskItem (path));
 						} else {
-							Log.LogWarningForXmlNode ("XA4218", ManifestFile, attribute, Properties.Resources.XA4218, path);
+							if (!bool.TryParse (required, out bool isRequired))
+								isRequired = true;
+							if (isRequired)
```

Comment:
This looks like the logic is backwards with the `!`, should it be:
```suggestion
							if (bool.TryParse (required, out bool isRequired) && isRequired)
```

Reply:
My thought was if "required" attribute  is "false" don't report, but if its "true" report it. If its missing the "required" attribute completely we need to report the warning as we cannot find the file.
`bool.TryParse` will return `false` if it fails.

Reply:
So, the default value of `required` is `true` if you leave it blank?
* https://developer.android.com/guide/topics/manifest/uses-library-element

That does seem to be what the docs say.

The only thing odd would be `required="banana"` would also show the warning, but you'd probably also get some other build error if you did that.

Reply:
I'm not sure what the default value for the element is. But in this case we are parsing the xml manually, and I didn't want to ignore warnings if something was missing.